model-optimizer/extensions/back/ChangeOutputTypeAttributes.py

```diff
@@ -73,17 +73,30 @@ def assert_that_is_castable_to_fp16(node: Node):
         val = node.in_port(i).data.get_value()
         if val is None:
             return
-
-        if np.any(val > np.finfo(np.float16).max) or np.any(val < np.finfo(np.float16).min):
-            raise Error("Try to convert with --data_type=FP32 argument. "
-                        "This model can not be converted to FP16 precision, since "
-                        "'{}' node value {} exceeds FP16 allowed limits: [{}, {}]"
-                        .format(node_name, val, np.finfo(np.float16).min, np.finfo(np.float16).max))
         # further this input values will be rewritten since force_shape_inference=True
         node.in_port(i).data.set_value(val.astype(np.float16))
```

Comment:
I agree that this looks tricky and complicated. But I didn't find easier way. Let me explain why I did this.

Previously we set only node attribute `node[dst_type] = final_type` (while `final_type = np.float16`). But for `Range` since it's attribute is rewritten by force during shape inference 
https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/ops/range.py#L74-L76
we need to ensure that it's not gonna happen that's why I also change inputs dtypes. 

Also I do shape inference for `Range` to check if limits are adequate. For `Cast` there is not need to check limits we only warn user if there is data loose during casting.

Yes, it looks complicated but I didn't found easier solution. 

But the good news is that after @mvafin changes #7588 this transformation will not be run (unless user wants convert the old way).